### PR TITLE
Save the highlighted item when switching tab

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
To be merged after #45281.

r? @rust-lang/docs

### CODE REVIEW

File: src/librustdoc/html/static/main.js [hunk 1]
```diff
@@ -696,39 +703,52 @@
             var search_input = document.getElementsByClassName('search-input')[0];
             search_input.onkeydown = null;
             search_input.onkeydown = function(e) {
-                var actives = [];
+                var actives = [[], [], []];
+                var current = 0;
```
Comment (dns2utf8) [inline src/librustdoc/html/static/main.js]: What does this `current` store? Is it the same as the `currentTab` from the outer scope?

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/main.js]: No, it's a local one when we're iterating over tabs.

File: src/librustdoc/html/static/main.js [hunk 2]
```diff
@@ -696,39 +703,52 @@
             var search_input = document.getElementsByClassName('search-input')[0];
             search_input.onkeydown = null;
```
Comment (dns2utf8) [inline src/librustdoc/html/static/main.js]: Is that line still required?

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/main.js]: Nope. Removing it. Thanks for spotting it!

**Review by dns2utf8 — APPROVED** [review]
Looks good to me. Thank you 👍

**Comment by QuietMisdreavus** [comment]
Looks good! r=me when https://github.com/rust-lang/rust/pull/45281 hits. This seems to be a superset of that PR, so hopefully it won't make a merge conflict.

**Pull request merged**